webapp/packages/core-ui/src/Hints/useHints.ts

```diff
@@ -0,0 +1,150 @@
+/*
+ * CloudBeaver - Cloud Database Manager
+ * Copyright (C) 2020-2023 DBeaver Corp and others
+ *
+ * Licensed under the Apache License, Version 2.0.
+ * you may not use this file except in compliance with the License.
+ */
+import { action, computed, observable } from 'mobx';
+import { useEffect, useRef } from 'react';
+import type { MenuStateReturn } from 'reakit';
+
+import { useObservableRef } from '@cloudbeaver/core-blocks';
+import type { SqlResultColumn } from '@cloudbeaver/core-sdk';
+import { type IDatabaseDataModel, type IDatabaseResultSet, ResultSetViewAction } from '@cloudbeaver/plugin-data-viewer';
+
+export interface Hint {
```

Comment:
please use autocompletion instead of hint and make global exported names more specific like `AutocompletionComponentCompletion`

and use `I` prefix for global exported interfaces